### PR TITLE
Fix copystep when run via gpup

### LINKED ISSUE #52: Plugin updating does not respect subfolders
Recently something has changed in the plugin update algorithm (at least in v1.4.3.0). It doesn't longer preserve sub-folders from the plugin packages and collapses all the files in the single root directory. 

While the problem itself is simple  it has very severe effect on plugins that do rely on file structure. Thus `CS-Script`  plugin was completely broken because of this problem.

### Test-case 
- Do clean install of `CS-Script (C# Intellisense)` plugin. 
Ignore prompt to install plugin internal update that tries to provide a work around for this problem. 
- Observe plugin flat file structure:
image

And this is the actual file structure of the plugin package:
image

### PR DESCRIPTION
**Pull request by bruderstein**
The CopyStep serialisation was using the wrong destination path for recursive directories, and using the wrong flag for validate.

Fixes #52

### CODE REVIEW

**Comment by bruderstein** [comment]
@chcg did you say you'd left a comment here?

Responding to the comment from the other issue... yes, it would be great to have unit tests (and ideally some integration tests) for all of the installation steps, but there's currently nothing in place for it. We'll need to think about mocking the filesystem, or injecting something to handle the file calls.

I'd like to get this out, and attack unit tests as a separate item - this is a pretty ugly bug.

File: libinstall/src/CopyStep.cpp [hunk 1]
```diff
@@ -323,21 +329,14 @@ StepStatus CopyStep::copyDirectory(tstring& fromPath, tstring& toPath,
...
 						if (_backup)
 							copyElement->SetAttribute(_T("backup"), _T("true"));
 						
-						if (_recursive) {
-							copyElement->SetAttribute(_T("recursive"), _T("true"));
-						}
-
 						if (_validate) {
-							copyElement->SetAttribute(_T("recursive"), _T("true"));
+							copyElement->SetAttribute(_T("validate"), _T("true"));
```
Comment (chcg) [inline libinstall/src/CopyStep.cpp]: Missed to hit submit, so it was not visible.

Comment (bruderstein) [inline libinstall/src/CopyStep.cpp]: Yeah, that was it. I recognised it as soon as I saw what was happening in the `PluginManagerGpup.xml` file.

Comment (chcg) [inline libinstall/src/CopyStep.cpp]: Otherwise the fix looks ok to me.

**Comment by chcg** [comment]
Do you think we can do a release now?

**Comment by bruderstein** [comment]
Yes, i think this should go out. I wanted to update the version and run some tests on upgrading tonight, then push the list to both locations. I can't do anything more until tonight, but if you can do the version update, that'd be great (no worries if not, I can do it tonight).

**Comment by chcg** [comment]
I will do it the version update now. Would else should I do on vacation go out into the sun and get burned ;-)

**Comment by bruderstein** [comment]
Whaaa... haha, no leave it, enjoy your vacation! I'll sort it tonight :)

**Comment by chcg** [comment]
Already tagged it and created the release, see https://github.com/bruderstein/nppPluginManager/releases/edit/v1.4.5. Already added some description before appveyor was ready, but that is gone now :-(

**Comment by bruderstein** [comment]
Devlist seems to be working now (i just ticked it and got the 1.4.5 version - it can take a few minutes for the changes to be applied.

Release looks great, thanks.

I've run through a clean install of N++ 7.3.2, updated to 1.4.3, then updated to 1.4.5 and it went smoothly. I'll update the old list so it shows the update directly to Plugin Manager 1.4.5, then after a few days I'll update it so that it only shows the new Plugin Manager, forcing people to update rather than stay on the old version (I don't want to keep maintaining the old list location, although I guess we'll have to for a while)

**Comment by bruderstein** [comment]
Done - all updated. It'll take 24 hours or so for the caches to update, so not everyone sees it immediately.

I've also update the list for 1.3.5 (and earlier), so they'll be updated to 1.4.5 directly too.

@chcg Thanks again for your help, enjoy your vacation!

**Pull request merged**